app/services/status_tag_service.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class StatusTagService < ::WasteCarriersEngine::BaseService
+  def run(resource:)
+    return [] if resource.blank?
+
+    @resource = resource
+
+    tags = []
+    tags << metadata_status
+    tags << pending_conviction_check
+    tags << pending_payment
+    tags << stuck
+
+    tags.compact
+  end
+
+  private
+
+  def metadata_status
+    return reg_metadata_status if @resource.is_a?(WasteCarriersEngine::Registration)
+    return transient_reg_metadata_status if @resource.is_a?(WasteCarriersEngine::TransientRegistration)
+  end
+
+  def reg_metadata_status
+    return :in_progress if @resource.metaData.PENDING?
+
+    @resource.metaData.status.downcase.to_sym
+  end
+
+  def transient_reg_metadata_status
+    return :revoked if @resource.metaData.REVOKED?
```

Comment:
Could the revoked and refused checks be moved to the first 2 lines of `metadata_status` as I think they are applicable to both registrations and transient registrations? Plus if the status is one of these it trumps all others.